### ISSUE TITLE
Mailbox UI inconsistencies when changing read status

### ISSUE THREAD

**Issue by eczarny** [elementary/mail#820]
### What Happened?

I've noticed quite a few UI inconsistencies with the mailbox when changing mail read status. Here's one example:

![image](https://user-images.githubusercontent.com/28012/217018636-a13b6f6e-6094-4298-b23c-74290cba1695.png)

The messages all have the subject "Unknown" with an empty timestamp. Opening a message (as shown in the screenshot) shows no content.

### Steps to Reproduce

1. Launch the Mail app
2. Select a message, then use the contextual menu to mark message as unread
3. After a few moments messages in the mailbox will disappear and be replaced by these "Unknown" messages.

### Expected Behavior

Messages should successfully have their read status changed without ruining the mailbox UI with inconsistencies.

### OS Version

7.x (Early Access)

### Software Version

Latest release (I have run all updates)

### Log Output

```shell
(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_uid: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_date_received: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_date_sent: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_uid: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_uid: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_date_received: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_date_sent: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_date_received: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_date_sent: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_uid: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_date_received: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_date_sent: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.369: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_subject: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_from: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_from: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.369: camel_message_info_get_subject: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.369: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.369: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.369: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.370: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.370: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.370: camel_message_info_get_subject: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.370: camel_message_info_get_from: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.370: camel_message_info_get_subject: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.370: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.370: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.370: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.371: camel_message_info_get_from: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.371: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.371: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.371: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.371: camel_message_info_get_subject: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.371: camel_message_info_get_from: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.371: camel_message_info_get_subject: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.371: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.371: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.371: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.371: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.372: camel_message_info_get_subject: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.372: camel_message_info_get_from: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): camel-CRITICAL **: 10:54:30.372: camel_message_info_get_subject: assertion 'CAMEL_IS_MESSAGE_INFO (mi)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.372: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.372: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): GLib-GObject-CRITICAL **: 10:54:30.372: g_object_get: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.mail:4770): Gtk-CRITICAL **: 10:54:30.373: _gtk_widget_get_preferred_size_for_size: assertion 'size >= -1' failed
[Thread 0x7fffc5f28640 (LWP 4804) exited]
[New Thread 0x7fffc5f28640 (LWP 4877)]
```

**Comment by eczarny** [comment]
I tried to run a bisect against 6.3.1 to identify the potential cause, but that release fails in a similar way. At this point I'm assuming there is some issue with the app's integration with libcamel1.2 on Jammy, but I haven't been able to confirm as I don't have a way to forcefully downgrade libcamel1.2 and its dependencies.

**Comment by lenemter** [comment]
Should be fixed by #825
